host-services/builtins/builtins_client.go

```diff
@@ -65,11 +73,15 @@ func (c *BuiltinServicesClient) KVSet(key string, value []byte) (*agentapi.HostS
 	return &kvResponse, nil
 }
 
-func (c *BuiltinServicesClient) KVDelete(key string) (*agentapi.HostServicesKeyValueResponse, error) {
+func (c *BuiltinServicesClient) KVDelete(key string, traceID *string) (*agentapi.HostServicesKeyValueResponse, error) {
```

Comment:
remove the trace ID from all of these functions.

Reply:
I'd imagine this should be `KVDelete(ctx context.Context, key string) (...)` and then when it calls the generic host services client `PerformRpc` that should also take a context, e.g. `PerformRpc(ctx context.Context, ...)` when perform RPC is crafting a messaging, it should then use the trace context Inject method to copy all relevant trace data from the context to the NATS headers.

We should not have a constant for the header key used for tracing, and should always use context round tripping and never manually read/write the traceparent header.

Using the context is idiomatic and it also allows us to add values and information later without changing the signature (this is why like 90% of Go libraries accept a context as the first method param)